Orochi/OrochiUtils.cpp

```diff
@@ -381,24 +381,45 @@ struct OrochiUtilsImpl
 	}
 };
 
-char* OrochiUtils::s_cacheDirectory = "./cache/";
+OrochiUtils::OrochiUtils() 
+{
+	m_cacheDirectory = "./cache/";
+}
 
-oroFunction OrochiUtils::getFunctionFromFile( oroDevice device, const char* path, const char* funcName, std::vector<const char*>* optsIn )
+OrochiUtils::~OrochiUtils() 
 {
-	const std::string cacheName = OrochiUtilsImpl::getCacheName( path, funcName );
+}
+
+oroFunction OrochiUtils::getFunctionFromFile( oroDevice device, const char* path, const char* funcName, std::vector<const char*>* optsIn )
+{ 
+	std::string cacheName = OrochiUtilsImpl::getCacheName( path, funcName );
+	if( m_kernelMap.find( cacheName.c_str() ) != m_kernelMap.end() )
+	{
+		return m_kernelMap[ cacheName ];
+	}
 
 	std::string source;
-	if( !OrochiUtilsImpl::readSourceCode( path, source, 0 ) ) return 0;
+	if( !OrochiUtilsImpl::readSourceCode( path, source, 0 ) )
+		return 0;
 
-	return getFunction( device, source.c_str(), path, funcName, optsIn );
+	oroFunction f = getFunction( device, source.c_str(), path, funcName, optsIn );
```

Comment:
Suggestion: we can also add a check to make sure it is not null before storing it.